packages/tc-ui/src/lib/components/structure.jsx

```diff
@@ -139,17 +139,10 @@ const MetaProperties = ({ data, isCreate }) => {
 	);
 };
 
-const LinkToRecord = ({
-	id,
-	type,
-	value: { name, code },
-	alternativeHostname,
-}) => {
-	const href = alternativeHostname
-		? `${alternativeHostname}/${encodeURIComponent(code)}`
-		: `/${type}/${encodeURIComponent(code)}`;
+const LinkToRecord = ({ id, type, value: { name, code }, linkGenerator }) => {
+	const href = linkGenerator({ type, code });
```

Comment:
```suggestion
	const href = linkGenerator({ type, ...value });
```
I think this is better as it gives more flexibility. It's also compatible with what you've already done.